sbot/timeout.py

```diff
@@ -1,28 +1,20 @@
 """Functions for killing the robot after a certain amount of time."""
+import _thread
 import logging
-import signal
-import sys
-from types import FrameType
+from threading import Timer
 
 logger = logging.getLogger(__name__)
 
-# TODO make this work on Windows
 
-
-def timeout_handler(signal_type: signal.Signals, stack_frame: FrameType) -> None:
+def timeout_handler() -> None:
     """
-    Handle the `SIGALRM` to kill the current process.
+    Handle the timeout to kill the current process.
 
     This function is called when the timeout expires and will stop the robot's main process.
     In order for this to work, any threads that are created must be daemons.
-
-    NOTE: This function is not called on Windows.
-
-    :param signal_type: The sginal that triggered this handler
-    :param stack_frame: The stack frame at the time of the signal
-    :raises SystemExit: To stop the robot's execution after the timeout
     """
-    raise SystemExit("Timeout expired: Game Over!")
+    logger.info("Timeout expired: Game Over!")
+    _thread.interrupt_main()
```

Comment:
Currently you will only generate a KeyboardInterrupt in the main thread which is easy to accidently catch in overly broad exeception handling. If we use SIGTERM it remains cross-platform but should generate a SystemExit instead. (You will also need `from signal import SIGTERM`)
```suggestion
    _thread.interrupt_main(SIGTERM)
```

Reply:
And, this only works in python 3.10. Back to the drawing board...

Reply:
TIL about the `signum` argument :bulb: 

You have to be pretty bad at exception handling to intercept `KeyboardInterrupt`. Notably, you have to either catch it explicitly, or catch `BaseException` (rather than merely `Exception`). But yes, `SIGINT` can be caught. I think I've found another solution...